core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/GetWalksHandler.java

```diff
@@ -0,0 +1,144 @@
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.data.Walk;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.data.EntitySeed;
+import uk.gov.gchq.gaffer.operation.impl.GetWalks;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.impl.output.ToEntitySeeds;
+import uk.gov.gchq.gaffer.operation.impl.output.ToVertices;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class GetWalksHandler implements OutputOperationHandler<GetWalks, Iterable<Walk>> {
+
+    private final Set<EntitySeed> visitedSeeds = new HashSet<>();
+    private int pathVertices;
+    private int hops;
+
+    @Override
+    public Iterable<Walk> doOperation(final GetWalks operation, final Context context, final Store store) throws OperationException {
+
+        // Check input
+        if (null == operation.getInput()) {
+            return null;
+        }
+
+        // Check operations input
+        final Iterable<GetElements> operations = operation.getOperations();
+
+        if (null == operations || Iterables.isEmpty(operations)) {
+            return new EmptyClosableIterable<>();
+        }
+
+        Iterable<? extends EntitySeed> seeds = operation.getInput();
+
+        hops = operation.getOperations().size();
+        pathVertices = hops + 1;
+
+        final HashBasedTable<Object, Object, Set<Edge>> subgraph = HashBasedTable.create();
+
+        // Execute the GetElements operations
+        for (final GetElements op : operations) {
+
+            final Set<EntitySeed> opSeeds = difference(visitedSeeds, Sets.newHashSet(seeds));
+
+            op.setInput(opSeeds);
+            final Iterable<Edge> results = (Iterable<Edge>) store.execute(op, context);
+
+            visitedSeeds.addAll(opSeeds);
+
+            // Cache results in HashBasedTable
+            for (final Edge e : results) {
+                if (null != subgraph.get(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue())) {
+                    final Set<Edge> set = subgraph.get(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue());
+                    set.add(e);
+                } else {
+                    final Set<Edge> set = new HashSet<>();
+                    set.add(e);
+                    subgraph.put(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue(), set);
+                }
+            }
+
+            final OperationChain<Iterable<? extends EntitySeed>> opChain = new OperationChain.Builder()
+                    .first(new ToVertices.Builder()
+                            .input(results)
```

Comment:
It looks like the `Iterable` `results` will be consumed twice causing the same edges to be pulled back from the database twice. Can this be avoided? The results are being cached in memory anyway. Maybe `results` should be cached?